streams/src/main/java/org/apache/kafka/streams/kstream/Consumed.java

```diff
@@ -96,13 +112,21 @@ protected Consumed(final Consumed<K, V> consumed) {
      *
      * @return a new instance of {@link Consumed}
      */
+    @Deprecated
     public static <K, V> Consumed<K, V> with(final Serde<K> keySerde,
                                              final Serde<V> valueSerde,
                                              final TimestampExtractor timestampExtractor,
                                              final Topology.AutoOffsetReset resetPolicy) {
         return new Consumed<>(keySerde, valueSerde, timestampExtractor, resetPolicy, null);
     }
 
+    public static <K, V> Consumed<K, V> with(final Serde<K> keySerde,
```

Comment:
Missing JavaDocs -- you can basically c&p the one from the existing `with()` method that we overload.

Reply:
Added docs to all, thank you